app/models/provider.coffee

```diff
@@ -34,4 +34,14 @@ Provider = NpiType.extend
     return (@get('otherNamePrefix') || '') + ' ' + (@get('otherFirstName') || '') + ' ' + (@get('otherMiddleName') || '') + ' ' + (@get('otherLastName') || '') + ' ' + (@get('otherNameSuffix') || '')
   ).property('otherNamePrefix', 'otherFirstName', 'otherMiddleName', 'otherLastName', 'otherNameSuffix')
 
+  classifcationSpecialties: ( ->
+    holder = []
+    taxonomies = @get('taxonomyLicenses')
+    taxonomies.forEach((license)->
+      classSpecialty = license.get('taxonomyCode').get('classificationSpecialty')
+      holder.push(classSpecialty) if classSpecialty not in holder
+    )
+    holder
```

Comment:
For code like this we should just pull in underscore.js (or lowdash), we can then simplify this to something like

    classSpecialties = _(@get('taxonomyLicenses')).map(...).uniq()

Actually, typing out loud here, does Ember offer similar stuff built in? 

http://emberjs.com/api/classes/Ember.Array.html#method_map

http://emberjs.com/api/classes/Ember.Array.html#method_uniq